# Importing modules, reading csv file and displaying first 5 records in the csv file.

In [1]:
import numpy as np
import pandas as pd
from datetime import datetime

In [2]:
airports = pd.read_csv('/Users/dukephung/Desktop/Spring2019/IS 362/' \
                       'nycflights13-master/data-raw/airports.csv',
                     usecols=[1, 2, 3])
airports.head()

name        lat        lon
0              Dillant Hopkins Airport  72.270833  42.898333
1           Eastport Municipal Airport  44.910111 -67.012694
2                         Houlton Intl  46.123083 -67.792056
3                         Caribou Muni  46.871500 -68.017917
4  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797

In [3]:
airports.set_index('name', inplace=True)

In [4]:
# airports.loc[:, 'lat'].max()
airports.head()

lat        lon
name                                                     
Dillant Hopkins Airport              72.270833  42.898333
Eastport Municipal Airport           44.910111 -67.012694
Houlton Intl                         46.123083 -67.792056
Caribou Muni                         46.871500 -68.017917
Northern Maine Rgnl At Presque Isle  46.688958 -68.044797

# What is the northernmost airport in the United States?

In [5]:
airports['lat'].head()

name
Dillant Hopkins Airport                72.270833
Eastport Municipal Airport             44.910111
Houlton Intl                           46.123083
Caribou Muni                           46.871500
Northern Maine Rgnl At Presque Isle    46.688958
Name: lat, dtype: float64

### Isolate the max latitude.

In [6]:
airports.loc[:,'lat'].max()


72.270833

### Perform a search within airports DataFrame using the max latitude as a filter.

In [7]:
airports[airports['lat'] == airports.loc[:,'lat'].max()]

lat        lon
name                                         
Dillant Hopkins Airport  72.270833  42.898333

### Because US in the northern hemisphere, latitude increases as you move further north.  Therefore to calculate the northernmost airport, I filtered the data to output the highest latitude value which proves it is the northern most airport.  Based on the results and assuming all data from the imported file is correct, Dillant Hopkins Airport is most north with a latitude of 72.27.

# 2.  What is the easternmost airport in the united states?

In [8]:
airports.head()

lat        lon
name                                                     
Dillant Hopkins Airport              72.270833  42.898333
Eastport Municipal Airport           44.910111 -67.012694
Houlton Intl                         46.123083 -67.792056
Caribou Muni                         46.871500 -68.017917
Northern Maine Rgnl At Presque Isle  46.688958 -68.044797

### The longitudinal values for the United states range in the negative value. This means that  any values in the longitude column may either be an error and reside in another country.  To begin the search for eastern most airport, a filter is used to limit the output from -30 to -70 in the 'lon' column.  

In [9]:
east_airports = airports[abs(airports['lon'] > -70)]
east_airports.count()

lat    15
lon    15
dtype: int64

In [10]:
east_airports = east_airports[abs(east_airports['lon'] < 0)]
east_airports

lat        lon
name                                                     
Eastport Municipal Airport           44.910111 -67.012694
Houlton Intl                         46.123083 -67.792056
Caribou Muni                         46.871500 -68.017917
Northern Maine Rgnl At Presque Isle  46.688958 -68.044797
Northern Aroostook Regional Airport  47.285556 -68.312778
Hancock County - Bar Harbor          44.449769 -68.361565
Banks Airport                        44.165389 -68.428167
Millinocket Muni                     45.647836 -68.685561
Bangor Intl                          44.807444 -68.828139
Knox County Regional Airport         44.060111 -69.099230
Augusta State                        44.320647 -69.797317

### Similar to the filter above, the highest negative value equates to eastern most airport while lowest equates to western most airport.  Using the filter on a newly created DataFrame object east_airports, the eastern most airport has been filtered out.  This airport is the Eastport Municipal Airport located in Maine.

In [11]:
east_airports[east_airports['lon'] == east_airports.loc[:,'lon'].max()]

lat        lon
name                                            
Eastport Municipal Airport  44.910111 -67.012694

# 3.  Which NY area airport had the windiest weather on February 12, 2013?

In [12]:
weather = pd.read_csv('/Users/dukephung/Desktop/Spring2019/IS 362/' \
                      'nycflights13-master/data-raw/weather.csv',
                     usecols=[0, 1, 2, 3, 9])
weather.head()

origin  year  month  day  wind_speed
0    EWR  2013      1    1    10.35702
1    EWR  2013      1    1     8.05546
2    EWR  2013      1    1    11.50780
3    EWR  2013      1    1    12.65858
4    EWR  2013      1    1    12.65858

### Filter the rows based on year, day and month of desired search.

In [13]:
weather_date = weather[weather['year'] == 2013]
weather_date = weather_date[weather_date['month'] == 2]
weather_date = weather_date[weather_date['day'] == 12]
weather_date.max()


origin            LGA
year             2013
month               2
day                12
wind_speed    1048.36
dtype: object

### Identify and remove all outliers

In [14]:
outliers = weather_date[weather_date['wind_speed'] > 150]
outliers

origin  year  month  day  wind_speed
1009    EWR  2013      2   12  1048.36058

### Results from output below indicates one outlier removed

In [15]:
weather_date = weather_date[weather_date['wind_speed'] < 150]
weather_date.count()

origin        71
year          71
month         71
day           71
wind_speed    71
dtype: int64

### Use .max() function to output the highest wind_speed from weather_date DataFrame.

In [16]:
weather_date.max()

origin            LGA
year             2013
month               2
day                12
wind_speed    23.0156
dtype: object

### The output from the previous line indicates that the origin airport, LGA, has the highest windspeed of 23.0156 on February 12, 2013.